### DISCUSSION TITLE
Only show deprecated API usage (KSV107)

### DISCUSSION THREAD

**Discussion by jkleinlercher** [aquasecurity/trivy#4782] [Q&A] [answered]
### Question

I want to use trivy to only show the usage of deprecated APIs.  There are some field in the json output I can filter on, but I ask myself if there is a way in trivy to filter on these misconfig types:

          "ID": "KSV107",
          "AVDID": "AVD-KSV-0107",
          "Title": "Evaluate k8s deprecated and removed APIs",
          "Namespace": "defsec.kubernetes.KSV107",
          "Query": "data.defsec.kubernetes.KSV107.deny",

Is there some builtin tool to filter on it? I saw there is a template feature but could you show me an example how I can filter on KSV107 in a template?

### Target

Kubernetes

### Scanner

Misconfiguration

### Output Format

None

### Mode

Client/Server

### Version

```bash
trivy --version
Version: 0.43.0
Vulnerability DB:
  Version: 2
  UpdatedAt: 2023-07-06 06:10:31.117953059 +0000 UTC
  NextUpdate: 2023-07-06 12:10:31.117952559 +0000 UTC
  DownloadedAt: 2023-07-06 06:27:30.833571981 +0000 UTC
Java DB:
  Version: 1
  UpdatedAt: 2023-07-03 00:53:46.661061023 +0000 UTC
  NextUpdate: 2023-07-06 00:53:46.661060523 +0000 UTC
  DownloadedAt: 2023-07-03 13:04:33.70268622 +0000 UTC
Policy Bundle:
  Digest: sha256:30ca89c908eac67d337f8b393262de10ccc2b4f486e7b47bf7828167ad6840b5
  DownloadedAt: 2023-07-06 08:35:03.448244202 +0000 UTC
```

**Comment by knqyf263** [comment]
Trivy is able to filter out results by finding IDs.
https://aquasecurity.github.io/trivy/v0.43/docs/configuration/filtering/#by-finding-ids

But your case looks like the other way around. You can probably make use of OPA.
https://aquasecurity.github.io/trivy/v0.43/docs/configuration/filtering/#by-open-policy-agent

I've never tried, but the rego policy would be as below.

```
ignore {
	input. AVDID != "AVD-KSV-0107"
}
```

**Reply by jkleinlercher** [reply, marked as answer]
thanks for your suggestions! unfortunately "trivy conf" doesn't allow the "--ignore-policy" flag but "trivy fs" does.

So this does the trick:

`trivy fs --ignore-policy only-deprecated.rego --scanners config .`

with this only-deprecated.rego:

```
package trivy

import data.lib.trivy

default ignore = false

ignore {
  input.AVDID != "AVD-KSV-0107"
}
```

**Reply by knqyf263** [reply]
You can track this issue about adding `--ignore-policy` support in `trivy config`.
https://github.com/aquasecurity/trivy/issues/3490
